kairos-server/src/l1_sync/event_manager.rs

```diff
@@ -0,0 +1,73 @@
+use std::sync::Arc;
+
+use casper_event_toolkit::fetcher::{Fetcher, Schemas};
+use casper_event_toolkit::metadata::CesMetadataRef;
+use casper_event_toolkit::rpc::client::CasperClient;
+
+use crate::state::ServerStateInner;
+use kairos_circuit_logic::transactions::{KairosTransaction, L1Deposit};
+
+use super::error::L1SyncError;
+
+pub struct EventManager {
+    next_event_id: u32,
+    fetcher: Fetcher,
+    schemas: Schemas,
+    server_state: Arc<ServerStateInner>,
+}
+
+impl EventManager {
+    pub async fn new(
+        rpc_url: &str,
+        contract_hash: &str,
+        server_state: Arc<ServerStateInner>,
+    ) -> Result<Self, L1SyncError> {
+        tracing::info!("Initializing event manager");
+
+        let client = CasperClient::new(rpc_url);
+        let metadata = CesMetadataRef::fetch_metadata(&client, contract_hash).await?;
+        tracing::debug!("Metadata fetched successfully");
+
+        let fetcher = Fetcher {
+            client,
+            ces_metadata: metadata,
+        };
+        let schemas = fetcher.fetch_schema().await?;
+        tracing::debug!("Schemas fetched successfully");
+
+        Ok(EventManager {
+            next_event_id: 0,
+            fetcher,
+            schemas,
+            server_state,
+        })
+    }
+
+    /// Processes new events starting from the last known event ID.
+    pub async fn process_new_events(&mut self) -> Result<(), L1SyncError> {
+        tracing::info!("Looking for new events");
+
+        let num_events = self.fetcher.fetch_events_count().await?;
+        for i in self.next_event_id..num_events {
+            let event = self.fetcher.fetch_event(i, &self.schemas).await?;
+            tracing::debug!("Event {} fetched: {:?}.", i, event);
+
+            // TODO: Parse full transaction data from event, then push it to Data Availability layer.
+
+            // TODO: Once we have ASN transaction, it should be converted and pushed into batch.
+            let recipient: Vec<u8> = "cafebabe".into();
```

Comment:
why are we mocking the deposit here?

Reply:
I was hoping that https://github.com/cspr-rad/kairos/pull/88 will get merged, and then it could be used for parsing. Since `kairos-tx` in the contract is postponed for later discussion, I will implement basic parsing here.

Reply:
#121 got merged, so I introduced `Deposit` event parsing in https://github.com/cspr-rad/kairos/pull/96/commits/ff7b6c69621e284696c10e460d8f35cdce9ae2b3.

However, `recipient` field is still mocked. It is not currently possible to obtain depositor **public key** from event, as it contains **account hash** (unnecessarily wrapped in `Key`)... This was already solved in [`kairos-tx` for contract PR](https://github.com/cspr-rad/kairos/pull/88/commits/75b5143602fa2bf3498044489b46e9d3277169c0#diff-a69ffc4ce0a99c02c5f8feb4f57529efd0c1871a711bad036378a977c1d9b2bfR55-R63) - validation if explicitly given public key matches with caller account hash - so maybe we should reconsider merging it before demo?